pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -631,4 +751,50 @@ public int hashCode() {
     result = EqualityUtils.hashCodeOf(result, _dateTimeFieldSpecs);
     return result;
   }
+
+  public boolean isVirtualColumn(String columnName) {
+    return columnName.startsWith("$") || (getFieldSpecFor(columnName).getVirtualColumnProvider() != null
+        && !getFieldSpecFor(columnName).getVirtualColumnProvider().isEmpty());
+  }
+
+
+  @JsonIgnore
+  public static byte[] getByteArrayFromField(Object value, DimensionFieldSpec fieldSpec) {
```

Comment:
why is this needed? it will be great if we can avoid static methods in config objects. We can probably move this SchemaUtil

Reply:
done.

It is used for converting a pinot column value into the appropriate binary representation for upsert primary key field. (make sure all primary key will become a byte array)